cli/src/main/java/com/devonfw/tools/ide/commandlet/FileExtractor/CommandletFileExtractorImpl.java

```diff
@@ -0,0 +1,157 @@
+package com.devonfw.tools.ide.commandlet.FileExtractor;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import com.devonfw.tools.ide.cli.CliException;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.io.FileAccess;
+import com.devonfw.tools.ide.io.TarCompression;
+import com.devonfw.tools.ide.process.ProcessContext;
+import com.devonfw.tools.ide.tool.ToolCommandlet;
+import com.devonfw.tools.ide.util.FilenameUtil;
+
+/**
+ * Implements common {@link CommandletFileExtractor} for a {@link ToolCommandlet}
+ */
+public class CommandletFileExtractorImpl implements CommandletFileExtractor {
+
+  protected final IdeContext context;
+
+  protected final ToolCommandlet commandlet;
+
+  /**
+   * The constructor
+   * 
+   * @param context the {@link IdeContext}.
+   * @param commandlet the {@link ToolCommandlet}
+   */
+  public CommandletFileExtractorImpl(IdeContext context, ToolCommandlet commandlet) {
+
+    this.context = context;
+    this.commandlet = commandlet;
+  }
+
+  @Override
+  public void extract(Path file, Path targetDir, boolean isExtract) {
+
+    FileAccess fileAccess = this.context.getFileAccess();
+    if (isExtract) {
+      Path tmpDir = fileAccess.createTempDir("extract-" + file.getFileName());
+      this.context.trace("Trying to extract the downloaded file {} to {} and move it to {}.", file, tmpDir, targetDir);
+      String extension = FilenameUtil.getExtension(file.getFileName().toString());
+      this.context.trace("Determined file extension {}", extension);
+      TarCompression tarCompression = TarCompression.of(extension);
+      ExtractorFileType fileType = ExtractorFileType.valueOf(extension.toUpperCase());
```

Comment:
this will throw an exception at runtime if the extension is not know here.
At least then the following code can never be reached anymore:
```
} else {
        throw new IllegalStateException("Unknown archive format " + extension + ". Can not extract " + file);
```